griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,211 @@
+            logging.error(e)
+            return ErrorArtifact(f"Error creating Google Doc: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to append text to a Google Doc",
+            "schema": Schema({"file_path": str, "text": str}),
+        }
+    )
+    def append_text_to_google_doc(self, file_path: str, text: str) -> InfoArtifact | ErrorArtifact:
+        try:
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            drive_service = self._build_client(self.DRIVE_SCOPES, "v3", "drive")
+            document_id = self._path_to_file_id(drive_service, file_path)
+            if document_id:
+                doc = service.documents().get(documentId=document_id).execute()
+                content = doc["body"]["content"]
+                last_text = content[-1]["paragraph"]["elements"][-1]["textRun"]["content"]
+                append_index = content[-1]["endIndex"]
+                if last_text.endswith("\n"):
+                    append_index -= 1
+
+                requests = [{"insertText": {"location": {"index": append_index}, "text": text}}]
+
+                doc = service.documents().batchUpdate(documentId=document_id, body={"requests": requests}).execute()
+                return InfoArtifact("Text appended successfully")
+            else:
+                return ErrorArtifact(f"Error: File not found for path {file_path}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error appending text to Google Doc: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to prepend text to a Google Doc",
+            "schema": Schema({"file_path": str, "text": str}),
+        }
+    )
+    def prepend_text_to_google_doc(self, file_path: str, text: str) -> InfoArtifact | ErrorArtifact:
+        try:
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            drive_service = self._build_client(self.DRIVE_SCOPES, "v3", "drive")
+            document_id = self._path_to_file_id(drive_service, file_path)
+            if document_id:
+                doc = service.documents().get(documentId=document_id).execute()
+
+                if len(doc["body"]["content"]) == 1:
+                    requests = [{"insertText": {"location": {"index": 1}, "text": text}}]
+                else:
+                    start_index = doc["body"]["content"][1]["startIndex"]
+                    requests = [{"insertText": {"location": {"index": start_index}, "text": text}}]
+
+                doc = service.documents().batchUpdate(documentId=document_id, body={"requests": requests}).execute()
+                return InfoArtifact("Text prepended successfully")
+            else:
+                return ErrorArtifact(f"Error: File not found for path {file_path}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error prepending text to Google Doc: {e}")
```

Comment:
Lowercase error string `error prepending text to Google Doc with path {file_path}: {e}`

Reply:
Done!